### PR TITLE
Getting started with Spring Boot social authentication in Kotlin

### LINKED ISSUE #1642: Getting started with Spring Boot social authentication in Kotlin
Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

### Introduction paragraph(s):
The Spring Framework is an application framework and inversion of control container for the JVM platform. In this tutorial the reader will learn how to authenticate users in a Spring Boot application through their social accounts i.e. Gmail, Facebook or GitHub account. 

### Key takeaways: 
1. The reader will learn the basics of Spring security.
2. The reader will be able to integrate Google, Facebook and GitHub authentication into their Spring Boot applications.

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.
[Spring Security Documentation](https://spring.io/projects/spring-security)

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by paulodhiambo**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you placed your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #1642

### CODE REVIEW

File: articles/spring-social-auth/index.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+        </dependency>
+```
+
+### Spring Security Configuration
+
+1. In the project's root package, create a package named `config`.
+2. Inside the `config` package created above, create a new kotlin file named `WebConfig` and add the code snippets below.
+
+```kotlin
+import org.springframework.context.annotation.Configuration
```
Comment (LinusMuema) [inline articles/spring-social-auth/index.md:67]: I think you can leave out the imports and when there is a chance of importing the wrong class, you can highlight it in the instructions. But they are not such great hinderance.

**Review by LinusMuema — APPROVED** [review]
Hi @paulodhiambo 
This is an awesome article. Just one comment to check for future articles.

Otherwise, everything is okay.

File: articles/spring-social-auth/index.md (hunk 1 above)
Comment (paulodhiambo) [reply articles/spring-social-auth/index.md:67]: @LinusMuema point noted.

**Comment by LinusMuema** [comment]
@hectorkambow 
You can give this article a final review.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
great work @paulodhiambo  👍  

cc @LinusMuema

**Pull request merged**